### PR TITLE
Support dark mode

### PR DESCRIPTION
**Pull request by xuanvan229**
Preview at https://federated-blog-ashy.vercel.app/

I used [next-theme](https://github.com/pacocoursey/next-themes#readme) because this
https://github.com/moaijs/moai/issues/63

Solve #14

### CODE REVIEW

File: components/Header/index.tsx
Comment (lqt93) [inline components/Header/index.tsx:31]: iconLabel is used for a11y, to inform user that this button is for changing theme. We are using Vietnamese, so in this case, I think it should be "Đổi ...".

**Review by lqt93 — CHANGES_REQUESTED** [review]
Except this, others are fine.

File: components/Header/index.tsx
Comment (lqt93) [inline components/Header/index.tsx:31]: IMO, it should be "Đổi chủ đề giao diện" for better meaning. If there is no "giao diện" in this context, user may misunderstand the meaning of button.

It maybe better if blind user cannot tab to focus on it because they do not need to change theme. But Idk if Moai support setting tabindex in button or not, please check.

@huytd can you give me your opinion in this case?

Comment (huytd) [reply components/Header/index.tsx:31]: IMHO, use explicit mode name for each case based on the `theme` variable, it would be:
- "Bật chế độ ban đêm" or "Dùng giao diện tối"
- "Bật chế độ ban ngày" or "Dùng giao diện sáng"

Comment (xuanvan229) [reply components/Header/index.tsx:31]: Sound good. I will do it

**Comment by huytd** [comment]
thanks @xuanvan229 , there is a small bug about dark mode does not shown when using the back button.

Steps to reproduce:

1. Go to home page
2. Click on an article to view
3. Switch to the other mode
4. Hit back button

Result: The theme switch back to the previous one

Expected: The current theme should stay the same when using back button

**Comment by huytd** [comment]
I have some fixes regarding the TS strict mode for Pagination, which will block this PR from building. Pls wait until this PR get merged https://github.com/webuild-community/federated-blog/pull/51

**Review by lqt93 — APPROVED** [review]
LGTM. Congratulation on your first PR for federated-blog 🥳

**Pull request merged**
